### PR TITLE
Add support for cookiecutter variables in strategies path

### PR DESCRIPTION
**Pull request by jpiron**
As this is supported by cookiecutter it should be supported by scaraplate
too.

### CODE REVIEW

File: src/scaraplate/config.py [hunk 1]
```diff
@@ -15,11 +15,34 @@ class StrategyNode(NamedTuple):
     config: Dict[str, Any]
 
 
-class ScaraplateYaml(NamedTuple):
-    default_strategy: StrategyNode
-    strategies_mapping: Mapping[str, StrategyNode]
-    git_remote_type: Optional[Type[GitRemote]]
-    cookiecutter_context_type: Type[CookieCutterContext]
+class ScaraplateYaml:
```
Comment (kxepal) [inline src/scaraplate/config.py]: `ScaraplateYaml` is sort of a dataclass. Let's keep it as `NamedTuple` for now. Or you see any reasons why it shouldn't be such?

Comment (KostyaEsmukov) [reply src/scaraplate/config.py]: @kxepal I believe it was turned to a class in order to allow mutation of the `strategies_mapping` property below.

Although I would prefer to keep it a NamedTuple as well.

**Comment by KostyaEsmukov** [comment]
There’s support for glob patterns in the strategies mapping, would that suffice for you? If no, could you please share your usecase?

In general I would prefer to not make the implementation more complex if the problem could be solved without introducing this feature.

File: src/scaraplate/config.py (hunk 1 above)
Comment (jpiron) [reply src/scaraplate/config.py]: Indeed I had to turn it into a class for mutating `strategies_mapping`

**Comment by jpiron** [comment]
Our main usecase is to combined it with the `IfNewProject` strategy.
Here is an excerpt of one of our cookiecutters : 
```
{{ cookiecutter.project_dest }}/
├── {{cookiecutter.project_slug}}
│   ├── __init__.py
│   ├── jobs
│   │   ├── {{cookiecutter.project_slug}}.py
│   │   └── __init__.py
│   ├── main.py
│   └── settings.py
```
We'd like the `{{cookiecutter.project_slug}}/jobs/{{cookiecutter.project_slug}}.py` and the `{{cookiecutter.project_slug}}/jobs/__init__.py` to be rendered only at project creation.

For the `__init__.py` module glob patterns could be used. But for the  `{{cookiecutter.project_slug}}.py` module, as the `jobs` package is very likely to contain other jobs modules glob patterns can't be used.

**Comment by KostyaEsmukov** [comment]
@jpiron I have just opened PR #13 which is an alternative implementation of this feature. I would appreciate it if you could have a look.

**Comment by jpiron** [comment]
Nice ! I didn't had the time to test it yet, but it seems a better implementation than mine. I'll close this PR in favor of yours.

**Comment by KostyaEsmukov** [comment]
Ok, thanks for looking into this! I'm going to release it in `0.2`, which should happen approximately in a week.

**Pull request closed without merging**